test/com/google/enterprise/adaptor/opentext/AuthenticationHandlerTest.java

```diff
@@ -61,6 +65,63 @@ public void testGetNewToken() throws SOAPException {
     assertEquals("incoming token", handler.getAuthenticationToken());
   }
 
+  @Test
+  public void testEmptyMessage() throws SOAPException {
+    SOAPMessageContextMock contextMock = new SOAPMessageContextMock(false);
+    contextMock.message = null;
+    SOAPMessageContext context =
+        Proxies.newProxyInstance(SOAPMessageContext.class, contextMock);
+    AuthenticationHandler handler = new AuthenticationHandler("authtoken");
+    List<String> messages = new ArrayList<>();
+    captureLogMessages(AuthenticationHandler.class,
+        "SOAPMessageContext.message==null", messages);
+    assertTrue(handler.handleMessage(context));
+    assertEquals(messages.toString(), 1, messages.size());
+  }
+
+  @Test
+  public void testEmptyHeaderInbound() throws SOAPException {
+    SOAPMessageContextMock contextMock = new SOAPMessageContextMock(false);
+    contextMock.message.getSOAPPart().getEnvelope().getHeader().detachNode();
+    assertEquals("header should be missing",
+        null, contextMock.message.getSOAPHeader());
+    SOAPMessageContext context =
+        Proxies.newProxyInstance(SOAPMessageContext.class, contextMock);
+    AuthenticationHandler handler = new AuthenticationHandler(null);
+    assertTrue(handler.handleMessage(context));
+    assertEquals(null, handler.getAuthenticationToken());
+  }
+
+  @Test
+  public void testEmptyHeaderOutbound() throws SOAPException {
+    SOAPMessageContextMock contextMock = new SOAPMessageContextMock(true);
+    contextMock.message.getSOAPPart().getEnvelope().getHeader().detachNode();
+    assertEquals("header should be missing",
+        null, contextMock.message.getSOAPHeader());
+    SOAPMessageContext context =
+        Proxies.newProxyInstance(SOAPMessageContext.class, contextMock);
+    AuthenticationHandler handler = new AuthenticationHandler("authtoken");
+    assertTrue(handler.handleMessage(context));
+    assertAuthenticationHeaderEquals("authtoken",
+        contextMock.message.getSOAPHeader());
+  }
+
+  // Test interface methods for coverage
+  @Test
+  public void testBasicMethods() throws SOAPException {
+    SOAPMessageContextMock contextMock = new SOAPMessageContextMock(false);
+    SOAPMessageContext context =
+        Proxies.newProxyInstance(SOAPMessageContext.class, contextMock);
+    AuthenticationHandler handler = new AuthenticationHandler("authtoken");
+    Set<QName> headers = handler.getHeaders();
+    assertEquals(1, headers.size());
```

Comment:
This could give a better message, through some combination of headers.toString() here, using assertEquals with an expected Set, or using a Hamcrest assertion like hasItem. The set equality seems like the best approach.